frontend/appflowy_flutter/lib/user/presentation/screens/sign_in_screen/widgets/third_party_sign_in_buttons.dart

```diff
@@ -30,89 +30,108 @@ class ThirdPartySignInButtons extends StatelessWidget {
-            labelText: LocaleKeys.signIn_LogInWithDiscord.tr(),
-            onPressed: () {
-              _signInWithDiscord(context);
-            },
-          ),
-        ],
+      return LayoutBuilder(
+        builder: (context, constraints) => Column(
+          children: [
+            _ThirdPartySignInButton(
+              buttonWidth: constraints.maxWidth,
+              isMobile: true,
+              icon: FlowySvgs.google_mark_xl,
+              labelText: LocaleKeys.signIn_LogInWithGoogle.tr(),
+              onPressed: () {
+                _signInWithGoogle(context);
+              },
+            ),
+            const SizedBox(height: 8),
+            _ThirdPartySignInButton(
+              buttonWidth: constraints.maxWidth,
+              isMobile: true,
+              icon: isDarkMode
+                  ? FlowySvgs.github_mark_white_xl
+                  : FlowySvgs.github_mark_black_xl,
+              labelText: LocaleKeys.signIn_LogInWithGithub.tr(),
+              onPressed: () {
+                _signInWithGithub(context);
+              },
+            ),
+            const SizedBox(height: 8),
+            _ThirdPartySignInButton(
+              buttonWidth: constraints.maxWidth,
+              isMobile: true,
+              icon: isDarkMode
+                  ? FlowySvgs.discord_mark_white_xl
+                  : FlowySvgs.discord_mark_blurple_xl,
+              labelText: LocaleKeys.signIn_LogInWithDiscord.tr(),
+              onPressed: () {
+                _signInWithDiscord(context);
+              },
+            ),
+          ],
+        ),
       );
     }
     // ThirdPartySignInButtons in desktop
-    return Column(
-      children: [
-        _ThirdPartySignInButton(
-          key: const Key('signInWithGoogleButton'),
-          isMobile: false,
-          icon: FlowySvgs.google_mark_xl,
-          labelText: LocaleKeys.signIn_LogInWithGoogle.tr(),
-          onPressed: () {
-            _signInWithGoogle(context);
-          },
-        ),
-        const SizedBox(height: 8),
-        _ThirdPartySignInButton(
-          isMobile: false,
-          icon: isDarkMode
-              ? FlowySvgs.github_mark_white_xl
-              : FlowySvgs.github_mark_black_xl,
-          labelText: LocaleKeys.signIn_LogInWithGithub.tr(),
-          onPressed: () {
-            _signInWithGithub(context);
-          },
-        ),
-        const SizedBox(height: 8),
-        _ThirdPartySignInButton(
-          isMobile: false,
-          icon: isDarkMode
-              ? FlowySvgs.discord_mark_white_xl
-              : FlowySvgs.discord_mark_blurple_xl,
-          labelText: LocaleKeys.signIn_LogInWithDiscord.tr(),
-          onPressed: () {
-            _signInWithDiscord(context);
-          },
-        ),
-      ],
+    return LayoutBuilder(
+      // Use LayoutBuilder to get the maxWidth of Column and pass it to _ThirdPartySignInButton to build the button with the same width.
```

Comment:
We could give them a maxwidth constraint that isn't dependant on window size, they grow huge if they just depend on the columns width constraints right?

Reply:
I just noticed the `AuthFormContainer` has already been limited to 340 on the desktop. 
<img width="622" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/400ef501-1d42-4f9e-9f3d-0de744dc9ae2">
Since the minimum window size is 800(greater than 340), no matter how we resize the window, it will always be 340 on the desktop. So I removed LayoutBuilder for the desktop part. 
For mobile, I chose to keep the LayoutBuilder there, because I worried that a fixed width may cause overflow in some narrow phone screens. Currently in mobile, the button width = screen size - 100(left+right padding).